Trimble.Modus.Components/Controls/DropDown/TMDropDown.cs

```diff
@@ -0,0 +1,346 @@
+            {
+                new ColumnDefinition { Width = GridLength.Star },
+                new ColumnDefinition { Width = GridLength.Auto }
+            },
+            RowDefinitions = new RowDefinitionCollection
+            {
+                new RowDefinition { Height = GridLength.Auto },
+            },
+            ColumnSpacing = 2,
+            HorizontalOptions = LayoutOptions.Center,
+            VerticalOptions = LayoutOptions.Center
+
+        };
+
+        grid.Add(label, 0, 0);
+        grid.Add(indicatorButton, 1, 0);
+        var stack = new StackLayout
+        {
+            Orientation = StackOrientation.Horizontal,
+            HorizontalOptions = LayoutOptions.Center,
+            VerticalOptions = LayoutOptions.Center,
+            Spacing = 2,
+        };
+        stack.Children.Add(label);
+        stack.Children.Add(indicatorButton);
+        innerBorder = new Border
+        {
+            Stroke = ResourcesDictionary.ColorsDictionary(ColorsConstants.TrimbleBlue),
+            StrokeThickness = 4,
+            BackgroundColor = ResourcesDictionary.ColorsDictionary(ColorsConstants.TrimbleBlue),
+            HeightRequest = 48,
+            WidthRequest = 240,
+            StrokeShape = new RoundRectangle
+            {
+                CornerRadius = new CornerRadius(4)
+            },
+            Content = stack,
+            HorizontalOptions = LayoutOptions.Start,
+            VerticalOptions = LayoutOptions.Start,
+
+            Padding = new Thickness(0)
+        };
+        var tapGestureRecognizer = new TapGestureRecognizer();
+        tapGestureRecognizer.Tapped += OnTapped;
+
+        innerBorder.GestureRecognizers.Add(tapGestureRecognizer);
+        indicatorButton.GestureRecognizers.Add(tapGestureRecognizer);
+        PopupService.Instance.Dismissed += OnPopupRemoved;
+        Content = innerBorder;
+    }
+    #endregion
+    #region Private Methods
+    private void OnSelected(object sender, SelectedItemChangedEventArgs e, ListView list)
+    {
+        if (SelectedItems != null)
+        {
+            SelectedItems.Clear();
+            SelectedItems.Add(e.SelectedItem);
+            UpdateCellColor(list);
+        }
+        label.Text = e.SelectedItem.ToString();
+        if (count > 0 && PopupService.Instance.PopupStack.Count > 0)
+        {
+
+            PopupService.Instance?.DismissAsync();
+            Close();
+        }
+        SelectedIndex = e.SelectedItemIndex;
+        count = 1;
+    }
+
+    private void OnTapped(object sender, EventArgs e)
+    {
+        isVisible = !isVisible;
+        if (isVisible)
+        {
+            Open();
+        }
+        else
+        {
+            Close();
+        }
+    }
+
+    private async void Close()
+    {
+        await Task.WhenAll(
+
+            indicatorButton.RotateTo(0, AnimationDuration)
+        );
+        isVisible = false;
+    }
+
+    private async void Open()
+    {
+        var dataTemplate = new DataTemplate(() =>
```

Comment:
It looks like you are creating all these UI components whenever the Open methods gets called. Can't we create these once and reuse again?

Reply:
For now components are recreated in open only in windows